contracts/facets/GBMFacet.sol

```diff
@@ -297,13 +368,106 @@ contract GBMFacet is IGBM, IERC1155TokenReceiver, IERC721TokenReceiver, Modifier
             s.auctions[_auctionId].bidMultiplier = s.initiatorInfo.bidMultiplier;
         }
 
+        if (_auctionPresetID != 0) {
+            s.auctions[_auctionId].owner = LibDiamond.contractOwner();
+            s.auctions[_auctionId].startTime = s.auctionPresets[_auctionPresetID].startTime;
+            s.auctions[_auctionId].endTime = s.auctionPresets[_auctionPresetID].endTime;
+            s.auctions[_auctionId].hammerTimeDuration = s.auctionPresets[_auctionPresetID].hammerTimeDuration;
+            s.auctions[_auctionId].bidDecimals = s.auctionPresets[_auctionPresetID].bidDecimals;
+            s.auctions[_auctionId].stepMin = s.auctionPresets[_auctionPresetID].stepMin;
+            s.auctions[_auctionId].incMin = s.auctionPresets[_auctionPresetID].incMin;
+            s.auctions[_auctionId].incMax = s.auctionPresets[_auctionPresetID].incMax;
+            s.auctions[_auctionId].bidMultiplier = s.auctionPresets[_auctionPresetID].bidMultiplier;
+
+            require(s.auctions[_auctionId].startTime != 0, "Auction startTime has not been set");
+        }
+
         //Event emitted when an auction is being setup
-        emit Auction_Initialized(_auctionId, _tokenId, _1155Index, _tokenContract, _tokenKind);
+        emit Auction_Initialized(_auctionId, _tokenId, _locIndex, _tokenContract, _tokenKind);
 
         //Event emitted when the start time of an auction changes (due to admin interaction )
         emit Auction_StartTimeUpdated(_auctionId, getAuctionStartTime(_auctionId));
     }
 
+	/// @notice Seller can cancel an auction during the grace period 
+    /// Throw if the token owner is not the caller of the function
+    /// @param _auctionID The auctionId of the auction to complete
+    function cancelAuction(uint256 _auctionID) public {
+		require(s.auctionSeller[_auctionID] != address(0), "Initial auctions cannot be cancelled");
+		
```

Comment:
Anybody can cancel another address's auction..there should be an ownership check

Reply:
Nice catch! @Nokhal is there any reason why we shouldn't limit permissions for this?

Reply:
No. Our design was that secondary market auction couldn't be cancelled (only primary one), so spec oversight on our side here when it come to cancelling secondary market auction.